lib/matplotlib/tests/test_axes.py

```diff
@@ -8272,3 +8272,15 @@ def test_bar_all_nan(fig_test, fig_ref):
 
     ax_ref.bar([1], [1]).remove()
     ax_ref.bar([1], [1])
+
+
+@image_comparison(["extent_units.pdf"], style="mpl20")
```

Comment:
I think we probably need somewhat more extensive testing than this - at least test that the x-axes works as expected?    Otherwise looks good.

Reply:
Hi, could you please let me know what as expected would mean here?

Reply:
You are only checking that dates work for the yaxis, not the xaxis

Reply:
ah okay, will add the checks